release_notes.md

```diff
@@ -159,10 +159,59 @@ User will get original byte now.
 
 # v1.0.0
 
-### Features
+The main goal of the release 1.0.0 is to implement new architecture that solves many of extension's problems. 
+It also allows you to create your own Commands/DataSources/SourceAPIs.
 
-1. [TH2-2465] Add module th2_gui_report to create a link by event id or message id
+## Attention
+
+The new version has raised the major version. It means the users should update their code to work with the new DS lib.
+
+Version 0.6.x won't have new features but will have bug fixes.
+
+Version 1.x.y doesn't use any adapters by default, 0.6.x does. If you need the save behavior as previous,
+use `CodecPipelinesAdapter` manually.
+
+## User impact and migration instructions
+
+1. [I] There is no more old-style DataSource class.  
+   [M] Use a specific class for your purposes (HTTPDataSource or GRPCDataSource).
+2. [I] There are no more `get_x` methods in the DataSource classes. Instead, it has the command method. This method
+   takes an object of the Command class.  
+   [M] So if you used `data_source.get_events_from_data_provider` use `data_source.command(GetEvents(ARGS))` instead.  
+   from `data_source.get_messages_from_data_provider` to `data_source.command(GetMessages(ARGS).apply_adapter(
+   codec_pipeline_adapter))`  
+   from `find_messages_by_id_from_data_provider` to `GetMessagesById` or `GetMessageById`  
+   from `find_events_by_id_from_data_provider` to `GetEventsById` or `GetEventById`  
+   if you used broken_events parameter initialize the command with use_stub=True parameter.
+3. [I] Modules structure was changed.  
+   [M] You need to change some import paths.
+4. [I] All class constructors/methods have got explicit arguments.  
+   [M] Change old args names to new if required.
+5. [I] Set of classes to create EventsTree representation was significantly changed.
+   [M] Use EventsTreesCollection instead of EventsTree. EventsTree is a real tree structure now.
```

Comment:
Should we indicate that the methods of EventsTree (in EventsTreesCollection) are changed too?

Reply:
You are right